### PR TITLE
multi: add control over maximum acceptable local CSV delay

### LINKED ISSUE #2426: Feature Request: Specify the lowest CSV delay we're willing to accept
Is there a option specifying the csv for autopilot to open channel?

Higher csv should means more time for disaster recovery.

### PR DESCRIPTION
**Pull request by carlaKC**
This PR provides nodes with a `MaxLocalCSV` config option which specifies the maximum delay that we will accept from the remote peer. This value allows users to set a maximum default value that suits their time preference. We do cap this value at `144` blocks at present, to ensure that users don't set an unreasonably low CSV which results in no channels ever succeeding.

For channels we open: this maximum can be set on a per-channel basis using `max_local_csv`, defaulting to our config option if this value is not set
For channels peers open: we will enforce the default value for all channels, and the `channelacceptor` can be used to take more custom actions. 

Fixes #2426

### CODE REVIEW

File: fundingmanager.go [hunk 1]
```diff
@@ -51,6 +51,11 @@ const (
 	minLtcRemoteDelay uint16 = 576
 	maxLtcRemoteDelay uint16 = 8064
 
+	// maxLocalCSVDelay is the maximum delay we accept on our commitment
+	// output.
+	// TODO(halseth): find a more scientific choice of value.
+	maxLocalCSVDelay = 10000
```
Comment (halseth) [inline fundingmanager.go:57]: rename to `defaultMaxLocalCSVDelay` and move to config

**Review by halseth — APPROVED** [review]
Nice addition, LGTM 👍

File: fundingmanager_test.go [hunk 2]
```diff
@@ -1273,6 +1275,158 @@ func TestFundingManagerNormalWorkflow(t *testing.T) {
...
+		return bobRequiredCSV
+	}
+
+	// For convenience, we bump our max pending channels to 2 so that we
+	// can test incoming and outgoing channels without needing to step
+	// through the full funding process.
+	alice.fundingMgr.cfg.MaxPendingChannels = 2
+	bob.fundingMgr.cfg.MaxPendingChannels = 2
+
+	// If our maximum exceeds the value bob sets, we expect this test to
```
Comment (bhandras) [inline fundingmanager_test.go:1312]: I think you wanted to write "our maximum is is less than"

File: fundingmanager_test.go [hunk 3]
```diff
@@ -1273,6 +1275,158 @@ func TestFundingManagerNormalWorkflow(t *testing.T) {
...
+	}
+
+	// Alice should have sent the OpenChannel message to Bob.
+	alice.fundingMgr.initFundingWorkflow(bob, initReq)
+	var aliceMsg lnwire.Message
+	select {
+	case aliceMsg = <-alice.msgChan:
+
+	case err := <-initReq.err:
+		t.Fatalf("error init funding workflow: %v", err)
```
Comment (bhandras) [inline fundingmanager_test.go:1335]: nit: could add a new line

File: lncfg/chain.go [hunk 4]
```diff
@@ -16,9 +20,28 @@ type Chain struct {
 
 	DefaultNumChanConfs int                 `long:"defaultchanconfs" description:"The default number of confirmations a channel must have before it's considered open. If this is not set, we will scale the value according to the channel size."`
 	DefaultRemoteDelay  int                 `long:"defaultremotedelay" description:"The default number of blocks we will require our channel counterparty to wait before accessing its funds in case of unilateral close. If this is not set, we will scale the value according to the channel size."`
+	MaxLocalDelay       uint16              `long:"maxlocaldelay" description:"The maximum blocks we will allow our funds to be timelocked before accessing its funds in case of unilateral close. If a peer proposes a value greater than this, we will reject the channel."`
```
Comment (bhandras) [inline lncfg/chain.go:23]: q: why not call this config value `MaxLocalCSVDelay`?

File: lncfg/chain.go [hunk 5]
```diff
@@ -16,9 +20,28 @@ type Chain struct {
...
 }
+
+// Validate performs validation on our chain config.
+func (c *Chain) Validate(minTimeLockDelta uint32, minDelay uint16) error {
+	if c.TimeLockDelta < minTimeLockDelta {
+		return fmt.Errorf("timelockdelta must be at least %v",
+			minTimeLockDelta)
+	}
+
+	// Check that our max local delay it isn't set below some reasonable
```
Comment (bhandras) [inline lncfg/chain.go:38]: nit: ~it~

File: cmd/lncli/cmd_open_channel.go [hunk 6]
```diff
@@ -135,6 +135,13 @@ var openChannelCommand = cli.Command{
 				"not set, we will scale the value according to the " +
 				"channel size",
 		},
+		cli.Uint64Flag{
+			Name: "max_local_csv",
+			Usage: "(optional) the maximum number of blocks that " +
+				"we will allow the remote peer to require we " +
+				"wait before accessing our fund in the case of " +
```
Comment (bhandras) [inline cmd/lncli/cmd_open_channel.go:142]: nit: funds

**Review by bhandras — APPROVED** [review]
LGTM, just a few nits :kangaroo:

**Comment by bhandras** [comment]
needs rebase

File: lncfg/chain.go (hunk 4 above)
Comment (carlaKC) [reply lncfg/chain.go:23]: I was following the naming for the existing `DefaultRemoteDelay`, since they're related.

**Pull request merged**